vagen/envs/webarena/browser_pool.py

```
@@ -0,0 +1,187 @@
+        await asyncio.gather(*(s.close() for s in self._slots), return_exceptions=True)
+        self._started = False
+
+    async def acquire_slot(self, timeout: float = 300.0) -> BrowserSlot:
+        """Reserve one context slot on the least-loaded browser."""
+        try:
+            await asyncio.wait_for(self._sem.acquire(), timeout=timeout)
+        except asyncio.TimeoutError:
+            raise RuntimeError(
+                f"BrowserPool acquire timed out after {timeout}s "
+                f"({self.stats_str()})"
+            )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)